service/history/hsm/registry.go

```diff
@@ -140,6 +143,27 @@ func RegisterImmediateExecutor[T Task](r *Registry, executor ImmediateExecutor[T
 	return nil
 }
 
+// RegisterRemoteMethod registers an [RemoteExecutor] for the given remote method definition.
+// Returns an [ErrDuplicateRegistration] if an executor for the type has already been registered.
```

Comment:
```suggestion
// Returns an [ErrDuplicateRegistration] if an executor for the method has already been registered.
```

Reply:
I just realized this isn't right. We should allow separate state machines to register the same method name.

Reply:
Registration should be done by state machine type + method name.

Reply:
I see this mistake was already in the other registration functions, and I missed this in a previous review.
I'll fix those later.